js/pillbox.js

```diff
@@ -298,48 +296,45 @@
 			}
 
 			if (items.length) {
-				$.each(items, function (i, item) {
+				var newItems = [];
+				$.each(items, function prepareItemForAdd (i, item) {
 					var $item = $(item.el);
-					var $neighbor;
 
 					$item.attr('data-value', item.value);
 					$item.find('span:first').html(item.text);
 
 					// DOM attributes
 					if (item.attr) {
-						$.each(item.attr, function (key, value) {
+						$.each(item.attr, function handleDOMAttributes (key, value) {
 							if (key === 'cssClass' || key === 'class') {
 								$item.addClass(value);
 							} else {
 								$item.attr(key, value);
 							}
 						});
-
 					}
 
 					if (item.data) {
 						$item.data('data', item.data);
 					}
 
-					$newHtml.push($item);
+					newItems.push($item);
```

Comment:
@cormacmccarthy if i'm reading this right each item is being updated and pushed onto another array. essentially a shallow clone? why not just `var newItems = items`?

for example `var newItems = []; var myItems = [{id: 1}, {id: 2}]; $.each(myItems, function(i, item) { item.attr = 'touch'; newItems.push(item); });`